File: javascript/reverse_string.js
function reverseString(str) {
  arr = str.split('')
  new_arr = []

  for(let i = arr.length - 1; i >=0; i--) {
    new_arr.push(arr[i])
  }

  return new_arr.join('')
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/reverse_string.rb
def reverse_string(str)
  arr = str.split('')
  new_arr = []
  i = arr.length - 1

  while i >= 0
    new_arr << arr[i]
    i -= 1
  end
  
  new_arr.join('')
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution